### PR TITLE
DRYing the build scripts

### PR DESCRIPTION
**Pull request by pwilkins**
Much of the build script is the same regardless of platform, so why repeat ourselves?  Let's reduce the effort of synchronizing our build scripts.  (I'm really tired of having one build script working and the others failing.  And it's never the same platform that works.)

Plan:
I will try to detect the OS in the script and see if that works.  Failing that, I'll separate out the base build steps from the platform specific ones.

@cjshawMIT @CLIxIndia

### CODE REVIEW

**Comment by cjshawMIT** [comment]
@pwilkins let me know when you think it's ready to merge.

**Comment by cjshawMIT** [comment]
@pwilkins you may want to check out PR #38. I was getting a "bad command" error on Windows when trying to build, so I had to update the build script and use a new `compile:ui:only` command...might affect your work on this PR.

**Comment by cjshawMIT** [comment]
@pwilkins regarding commit 06036ef270728e4d6cab0597fa263ab834ab406a, if you move the filesystem commands out, I think you need to either 1) update the README on how to update the UI files for dev / testing (because devs will have to manually copy the files over), or 2) provide alternate commands for convenience...

**Comment by pwilkins** [comment]
This build script is ready to test.

What it does:
1. It runs on any operating system (as long as they're Windows, Ubuntu, or OSX) and performs the same build steps.
2. It detects the OS and performs the platform specific operations.
3. It uses ``yarn`` instead of ``npm``.

Notes:
1. It still uses the ``nose`` tests.  I didn't make the simple edit to switch to ``pytest``.  :-(
2. It doesn't catch errors and fail gracefully.
3. It is written in ``bash`` script which works perfectly but is, well, ``bash`` script.

@cjshawMIT @CLIxIndia

File: scripts/build_scripts/build_script_all_ssl.sh [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+fi
+
+# update the virtual environment
+pip install -r $BUILD_ROOT/requirements.txt
+pip install -r $BUILD_ROOT/test_requirements.txt
+
+# generate the UI
+cd $BUILD_ROOT/ui
+
+case $UN2_BUILD_OS in
```
Comment (cjshawMIT) [inline scripts/build_scripts/build_script_all_ssl.sh]: Why is this here? The Windows and non-Windows scripts look the same to me...

File: scripts/build_scripts/build_script_all_ssl.sh [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    'windows')
+        echo Compiling only UI for Windows...
+        cd $BUILD_ROOT/ui
+        yarn install
```
Comment (cjshawMIT) [inline scripts/build_scripts/build_script_all_ssl.sh]: Have you tried running this on Windows? `yarn` didn't work for me on version `0.19.1`, but they've published more versions since then...

File: scripts/build_scripts/build_script_all_ssl.sh [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+cd $BUILD_ROOT/tool-repos
+
+# Content player
+if [ ! -d content_player ]
+then
+  git clone git@github.com:CLIxIndia-Dev/content_player.git
+  cp content_player/.env.example content_player/.env
+fi
+cd content_player
+git checkout release
```
Comment (cjshawMIT) [inline scripts/build_scripts/build_script_all_ssl.sh]: This process has changed...please see:

https://github.com/CLIxIndia-Dev/unplatform_v2/blob/master/scripts/build_scripts/build_script_win32_ssl.sh#L119-L130

File: scripts/build_scripts/build_script_all_ssl.sh (hunk 1 above)
Comment (pwilkins) [inline scripts/build_scripts/build_script_all_ssl.sh]: We have in the past treated this differently in Windows than in other platforms.  I did this so that we're prepared if we treated them differently again in the future.  I could simple take it out, since it isn't needed currently.

File: scripts/build_scripts/build_script_all_ssl.sh [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+# OEA Open Embedded Assessments
+cd ..
+
+if [ ! -d "OpenAssessmentsClient" ]
+then
+  git clone git@github.com:CLIxIndia-Dev/OpenAssessmentsClient.git
+fi
+cd OpenAssessmentsClient
+git checkout release
```
Comment (cjshawMIT) [inline scripts/build_scripts/build_script_all_ssl.sh]: This process has changed...please see:

https://github.com/CLIxIndia-Dev/unplatform_v2/blob/master/scripts/build_scripts/build_script_win32_ssl.sh#L142-L161

File: scripts/build_scripts/build_script_all_ssl.sh [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+        # TODO: When we build working QBank-lite binaries for OSX, change "ubuntu" to the "osx" slug
+        QBANK_FILE=$(find $BUILD_ROOT/tool-repos/qbank-lite-bundles/release/ -name qbank-lite*ubuntu* | sort -n | tail -1)
+        ;;
+esac
+echo QBank-lite binary is $QBANK_FILE
+cp $QBANK_FILE  $BUILD_ROOT/bundle/
+
+# Zip up the final bundle/ directory and name the file per the unplatform version
+VERSION=$(python -c "import json; print json.load(open('package.json', 'rb'))['version']")
+OUTPUT="unplatform_v${VERSION//\'/}_osx.zip"
```
Comment (cjshawMIT) [inline scripts/build_scripts/build_script_all_ssl.sh]: The `.zip` file name should also depend on OS.

File: scripts/build_scripts/build_script_all_ssl.sh (hunk 1 above)
Comment (pwilkins) [inline scripts/build_scripts/build_script_all_ssl.sh]: You'll also notice that I'm calling a different ``package.json`` so that the directory wrangling is done in one place, the build script.

File: scripts/build_scripts/build_script_all_ssl.sh (hunk 2 above)
Comment (pwilkins) [inline scripts/build_scripts/build_script_all_ssl.sh]: I'm running Yarn 21.3 and have had no problems on Windows.

File: scripts/build_scripts/build_script_all_ssl.sh (hunk 3 above)
Comment (pwilkins) [inline scripts/build_scripts/build_script_all_ssl.sh]: Will update.

File: scripts/build_scripts/build_script_all_ssl.sh (hunk 4 above)
Comment (pwilkins) [inline scripts/build_scripts/build_script_all_ssl.sh]: Will update.

File: scripts/build_scripts/build_script_all_ssl.sh [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+
+# generate the UI
+cd $BUILD_ROOT/ui
+
+case $UN2_BUILD_OS in
+    'windows')
+        echo Compiling only UI for Windows...
+        cd $BUILD_ROOT/ui
+        yarn install
+        yarn run compile:prod
```
Comment (cjshawMIT) [inline scripts/build_scripts/build_script_all_ssl.sh]: Where is this command in `package.json`? I only see `compile:ui` and `compile:ui:only`...
